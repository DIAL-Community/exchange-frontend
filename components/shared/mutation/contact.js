import { gql } from '@apollo/client'

export const DELETE_CONTACT = gql`
  mutation DeleteContact($id: ID!) {
    deleteContact(id: $id) {
      contact {
        id
        slug
        name
      }
      errors
    }
  }
`

export const CREATE_CONTACT = gql`
  mutation CreateContact(
    $name: String!
    $slug: String!
    $email: String!
    $title: String!
  ) {
    createContact(
      name: $name
      slug: $slug
      email: $email
      title: $title
    ) {
      contact {
        id
        name
        slug
        email
        title
      }
      errors
    }
  }
`
