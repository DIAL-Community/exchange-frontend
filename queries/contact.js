import { gql } from '@apollo/client'

export const CONTACT_SEARCH_QUERY = gql`
  query Contacts($search: String!) {
    contact(search: $search) {
      id
      name
      slug
      email
      title
    }
  }
`
