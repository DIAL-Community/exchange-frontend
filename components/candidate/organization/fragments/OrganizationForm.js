import React, { useState, useCallback, useContext, useRef } from 'react'
import { useRouter } from 'next/router'
import { useMutation } from '@apollo/client'
import { useIntl } from 'react-intl'
import { FaSpinner } from 'react-icons/fa6'
import { Controller, useForm } from 'react-hook-form'
import ReCAPTCHA from 'react-google-recaptcha'
import { ToastContext } from '../../../../lib/ToastContext'
import { useUser } from '../../../../lib/hooks'
import Input from '../../../shared/form/Input'
import UrlInput from '../../../shared/form/UrlInput'
import Checkbox from '../../../shared/form/Checkbox'
import ValidationError from '../../../shared/form/ValidationError'
import { HtmlEditor } from '../../../shared/form/HtmlEditor'
import { CREATE_CANDIDATE_ORGANIZATION } from '../../../shared/mutation/candidateOrganization'
import { Loading, Unauthorized } from '../../../shared/FetchStatus'
import { DEFAULT_PAGE_SIZE } from '../../../utils/constants'
import {
  CANDIDATE_ORGANIZATION_PAGINATION_ATTRIBUTES_QUERY,
  PAGINATED_CANDIDATE_ORGANIZATIONS_QUERY
} from '../../../shared/query/candidateOrganization'

const OrganizationForm = React.memo(({ organization }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const captchaRef = useRef()
  const [captchaValue, setCaptchaValue] = useState()

  const slug = organization?.slug ?? ''

  const { user, loadingUserSession } = useUser()

  const [mutating, setMutating] = useState(false)
  const [reverting, setReverting] = useState(false)

  const { showSuccessMessage, showFailureMessage } = useContext(ToastContext)

  const router = useRouter()
  const { locale } = router

  const [updateOrganization, { reset }] = useMutation(CREATE_CANDIDATE_ORGANIZATION, {
    refetchQueries: [{
      query: CANDIDATE_ORGANIZATION_PAGINATION_ATTRIBUTES_QUERY,
      variables: { search: '' }
    }, {
      query: PAGINATED_CANDIDATE_ORGANIZATIONS_QUERY,
      variables: { search: '', limit: DEFAULT_PAGE_SIZE, offset: 0 }
    }],
    onCompleted: (data) => {
      const { createCandidateOrganization: response } = data
      if (response.candidateOrganization && response.errors.length === 0) {
        const redirectPath = `/${locale}` +
                             `/candidate/organizations/${response.candidateOrganization.slug}`
        const redirectHandler = () => router.push(redirectPath)
        showSuccessMessage(
          format('toast.submit.success', { entity: format('ui.candidateOrganization.label') }),
          redirectHandler
        )
      } else {
        showFailureMessage(format('toast.submit.failure', { entity: format('ui.candidateOrganization.label') }))
        setMutating(false)
        reset()
      }
    },
    onError: () => {
      showFailureMessage(format('toast.submit.failure', { entity: format('ui.candidateOrganization.label') }))
      setMutating(false)
      reset()
    }
  })

  const [submitter] = organization ? organization.contacts : []

  const {
    handleSubmit,
    register,
    control,
    formState: { errors }
  } = useForm({
    mode: 'onSubmit',
    reValidateMode: 'onChange',
    shouldUnregister: true,
    defaultValues: {
      organizationName: organization?.name ?? '',
      website: organization?.website ?? '',
      description: organization?.description ?? '',
      createStorefront: organization?.createStorefront ?? false,
      name: submitter?.name ?? '',
      email: submitter?.email ?? '',
      title: submitter?.title ?? ''
    }
  })

  const doUpsert = async (data) => {
    if (user) {
      // Set the loading indicator.
      setMutating(true)
      // Pull all needed data from session and form.
      const { userEmail, userToken } = user
      const {
        website,
        description,
        organizationName,
        createStorefront,
        name,
        email,
        title
      } = data
      // Send graph query to the backend. Set the base variables needed to perform update.
      const variables = {
        slug,
        website,
        description,
        organizationName,
        createStorefront,
        name,
        email,
        title,
        captcha: captchaValue
      }

      updateOrganization({
        variables,
        context: {
          headers: {
            'Accept-Language': locale,
            'Authorization': `${userEmail} ${userToken}`
          }
        }
      })
    }
  }

  const updateCaptchaData = (value) => {
    setCaptchaValue(value)
  }

  const cancelForm = () => {
    if (organization) {
      setReverting(true)
      router.push(`/${locale}/candidate/organizations/${slug}`)
    }
  }

  return loadingUserSession
    ? <Loading />
    : user
      ? <form onSubmit={handleSubmit(doUpsert)}>
        <div className='px-4 lg:px-0 py-4 lg:py-6 text-dial-plum'>
          <div className='flex flex-col gap-y-6 text-sm'>
            <div className='text-xl font-semibold'>
              {organization
                ? format('app.editEntity', { entity: organization.name })
                : `${format('app.createNew')} ${format('ui.organization.label')}`}
            </div>
            <div className='flex flex-col gap-y-2'>
              <label className='text-dial-sapphire required-field' htmlFor='organizationName'>
                {format('ui.candidateOrganization.organizationName')}
              </label>
              <Input
                id='organizationName'
                {...register('organizationName', { required: format('validation.required') })}
                placeholder={format('ui.candidateOrganization.organizationName.placeholder')}
                isInvalid={errors.organizationName}
              />
              {errors.organizationName && <ValidationError value={errors.organizationName?.message} />}
            </div>
            <div className='flex flex-col gap-y-2'>
              <label className='text-dial-sapphire required-field' htmlFor='website'>
                {format('organization.website')}
              </label>
              <Controller
                name='website'
                control={control}
                render={({ field: { value, onChange } }) => (
                  <UrlInput
                    id='website'
                    value={value}
                    onChange={onChange}
                    isInvalid={errors.website}
                    placeholder={format('ui.candidateOrganization.website.placeholder')}
                  />
                )}
                rules={{ required: format('validation.required') }}
              />
              {errors.website && <ValidationError value={errors.website?.message} />}
            </div>
            <div className='flex flex-col gap-y-2'>
              <label className='text-dial-sapphire required-field'>
                {format('organization.description')}
              </label>
              <Controller
                name='description'
                control={control}
                render={({ field: { value, onChange } }) => (
                  <HtmlEditor
                    editorId='description-editor'
                    onChange={onChange}
                    initialContent={value}
                    placeholder={format('ui.candidateOrganization.description.placeholder')}
                    isInvalid={errors.description}
                  />
                )}
                rules={{ required: format('validation.required') }}
              />
              {errors.description && <ValidationError value={errors.description?.message} />}
            </div>
            <label className='flex gap-x-2 items-center self-start text-dial-sapphire'>
              <Checkbox {...register('createStorefront')} />
              {format('ui.candidateOrganization.createStorefront')}
            </label>
            <hr className='border-b border-dial-blue-chalk my-3' />
            <div className='flex flex-col gap-y-2'>
              <label className='required-field' htmlFor='name'>
                {format('ui.candidateOrganization.name')}
              </label>
              <Input
                id='name'
                {...register('name', { required: format('validation.required') })}
                placeholder={format('ui.candidateOrganization.name.placeholder')}
                isInvalid={errors.name}
              />
              {errors.name && <ValidationError value={errors.name?.message} />}
            </div>
            <div className='flex flex-col gap-y-2'>
              <label className='required-field' htmlFor='email'>
                {format('ui.candidateOrganization.email')}
              </label>
              <Input
                id='email'
                {...register('email', { required: format('validation.required') })}
                placeholder={format('ui.candidateOrganization.email.placeholder')}
                isInvalid={errors.email}
              />
              {errors.email && <ValidationError value={errors.email?.message} />}
            </div>
            <div className='flex flex-col gap-y-2'>
              <label htmlFor='title'>
                {format('ui.candidateOrganization.title')}
              </label>
              <Input
                id='title'
                {...register('title')}
                placeholder={format('ui.candidateOrganization.title.placeholder')}
              />
            </div>
            <ReCAPTCHA
              sitekey={process.env.NEXT_PUBLIC_CAPTCHA_SITE_KEY}
              onChange={updateCaptchaData}
              ref={captchaRef}
            />
            <div className='flex flex-wrap text-base mt-6 gap-3'>
              <button
                type='submit'
                className='submit-button'
                disabled={mutating || reverting || !captchaValue}
              >
                {`${format('app.submit')} ${format('ui.organization.label')}`}
                {mutating && <FaSpinner className='spinner ml-3' />}
              </button>
              {organization &&
                <button
                  type='button'
                  className='cancel-button'
                  disabled={mutating || reverting}
                  onClick={cancelForm}
                >
                  {format('app.cancel')}
                  {reverting && <FaSpinner className='spinner ml-3' />}
                </button>
              }
            </div>
          </div>
        </div>
      </form>
      : <Unauthorized />
})

OrganizationForm.displayName = 'OrganizationForm'

export default OrganizationForm
