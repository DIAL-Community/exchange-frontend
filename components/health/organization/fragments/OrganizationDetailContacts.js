import { useIntl } from 'react-intl'
import { useState, useCallback, useContext } from 'react'
import { useRouter } from 'next/router'
import { useMutation } from '@apollo/client'
import { Controller, useForm } from 'react-hook-form'
import { validate } from 'email-validator'
import { UPDATE_ORGANIZATION_CONTACTS } from '../../../shared/mutation/organization'
import { DisplayType } from '../../../utils/constants'
import { useUser } from '../../../../lib/hooks'
import { ToastContext } from '../../../../lib/ToastContext'
import ContactCard from '../../../contact/ContactCard'
import Input from '../../../shared/form/Input'
import ValidationError from '../../../shared/form/ValidationError'
import EditableSection from '../../../shared/EditableSection'
import Pill from '../../../shared/form/Pill'

const inputSectionStyle = 'flex flex-col gap-y-3'

const OrganizationDetailContacts = ({ organization, canEdit, headerRef }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const [contacts, setContacts] = useState(organization?.contacts)
  const [isDirty, setIsDirty] = useState(false)

  const { user } = useUser()
  const { locale } = useRouter()
  const { showSuccessMessage, showFailureMessage } = useContext(ToastContext)

  const [updateOrganizationContacts, { loading }] = useMutation(UPDATE_ORGANIZATION_CONTACTS, {
    onError() {
      setIsDirty(false)
      setContacts(organization.contacts)
      showFailureMessage(format('toast.submit.failure', { entity: format('ui.contact.header') }))
      reset()
    },
    onCompleted: (data) => {
      const { updateOrganizationContacts: response } = data
      if (response?.organization && response?.errors?.length === 0) {
        setIsDirty(false)
        setContacts(response?.organization?.contacts)
        showSuccessMessage(format('toast.submit.success', { entity: format('ui.contact.header') }))
      } else {
        setIsDirty(false)
        setContacts(organization.contacts)
        showFailureMessage(format('toast.submit.failure', { entity: format('ui.contact.header') }))
        reset()
      }
    }
  })

  const isContactNameUnique =
    (name) =>
      !contacts.some(contact => contact.name.toLocaleLowerCase() === name.toLocaleLowerCase())

  const { handleSubmit, control, reset, formState: { errors } } = useForm({
    mode: 'onSubmit',
    reValidateMode: 'onChange',
    shouldUnregister: true,
    defaultValues: {
      name: '',
      email: '',
      title: ''
    }
  })

  const addContact = (contact) => {
    reset()
    setIsDirty(true)
    setContacts([...contacts,  contact ])
  }

  const removeContact = (contact) => {
    setContacts([...contacts.filter(({ name }) => name !== contact.name)])
    setIsDirty(true)
  }

  const onCancel = () => {
    setContacts(organization.contacts)
    setIsDirty(false)
  }

  const onSubmit = () => {
    if (user) {
      const { userEmail, userToken } = user

      updateOrganizationContacts({
        variables: {
          slug: organization.slug,
          contacts
        },
        context: {
          headers: {
            'Accept-Language': locale,
            Authorization: `${userEmail} ${userToken}`
          }
        }
      })
    }
  }

  const displayModeBody = organization && contacts.length > 0
    ? <div className='grid grid-cols-1 lg:grid-cols-2 gap-x-8 gap-y-3'>
      {contacts.map((contact, index) =>
        <div key={`project-${index}`}>
          <ContactCard contact={contact} displayType={DisplayType.SMALL_CARD} />
        </div>
      )}
    </div>
    :  <div className='text-sm text-dial-stratos'>
      {format( 'ui.common.detail.noData', {
        entity: format('ui.contact.label'),
        base: format('ui.organization.label')
      })}
    </div>

  const sectionHeader =
    <div className='text-xl font-semibold text-health-blue' ref={headerRef}>
      {format('ui.contact.header')}
    </div>

  const editModeBody =
    <div className='px-4 lg:px-6 py-4 flex flex-col gap-y-3 text-sm'>
      <form onSubmit={handleSubmit(addContact)}>
        <div className='flex flex-col gap-y-3'>
          <label className={inputSectionStyle}>
            <div className='text-dial-sapphire required-field'>
              {format('ui.contact.name.label')}
            </div>
            <Controller
              name='name'
              control={control}
              render={({ field }) => (
                <Input {...field} isInvalid={errors.name} />
              )}
              rules={{
                required: format('validation.required'),
                validate: name => isContactNameUnique(name) || format('organization.validation.contact.uniqueName')
              }}
            />
            {errors.name && <ValidationError value={errors.name?.message} />}
          </label>
          <label className={inputSectionStyle}>
            <div className='text-dial-sapphire required-field'>
              {format('ui.contact.email.label')}
            </div>
            <Controller
              name='email'
              control={control}
              render={({ field }) => (
                <Input {...field} isInvalid={errors.email} />
              )}
              rules={{
                required: format('validation.required'),
                validate: value => validate(value) || format('validation.email')
              }}
            />
            {errors.email && <ValidationError value={errors.email?.message} />}
          </label>
          <label className={inputSectionStyle}>
            <div>
              {format('ui.contact.title.label')}
            </div>
            <Controller
              name='title'
              control={control}
              render={({ field }) => (
                <Input {...field} />
              )}
            />
          </label>
          <div className='flex my-3'>
            <button type='submit' className='submit-button'>
              {format('app.assign')}
            </button>
          </div>
        </div>
        <div className='flex flex-wrap gap-3'>
          {contacts.map((contact, contactIdx) => (
            <Pill
              key={contactIdx}
              label={`${format('ui.contact.name.label')}: ${contact.name}`}
              onRemove={() => removeContact(contact)}
            />
          ))}
        </div>
      </form>
    </div>

  return (
    <EditableSection
      canEdit={canEdit}
      sectionHeader={sectionHeader}
      onSubmit={onSubmit}
      onCancel={onCancel}
      isDirty={isDirty}
      isMutating={loading}
      displayModeBody={displayModeBody}
      editModeBody={editModeBody}
    />
  )
}

export default OrganizationDetailContacts
