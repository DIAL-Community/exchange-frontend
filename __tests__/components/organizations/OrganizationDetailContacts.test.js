import { fireEvent, screen } from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import OrganizationDetailContacts from '../../../components/organizations/OrganizationDetailContacts'
import { CONTACT_SEARCH_QUERY } from '../../../queries/contact'
import { render } from '../../test-utils'
import CustomMockedProvider, { generateMockApolloData } from '../../utils/CustomMockedProvider'
import { mockNextUseRouter } from '../../utils/nextMockImplementation'
import { contacts } from './data/OrganizationDetailContacts'
import { organization } from './data/OrganizationForm'

mockNextUseRouter()
describe('Unit test for the OrganizationDetailContacts component.', () => {
  const EDIT_BUTTON_TEST_ID = 'edit-button'
  const PILL_TEST_ID = 'pill'
  const PILL_REMOVE_BUTTON_TEST_ID = 'remove-button'
  const ASSIGN_BUTTON_TEST_ID = 'assign-button'
  const NAME_INPUT_TEST_ID = 'name-input'
  const EMAIL_INPUT_TEST_ID = 'email-input'
  const TITLE_INPUT_TEST_ID = 'title-input'

  const mockContacts = generateMockApolloData(CONTACT_SEARCH_QUERY, { search: '' }, null, contacts)

  test('Should assign button have text "Assign"', () => {
    const { container, getByTestId } = render(
      <CustomMockedProvider mocks={[mockContacts]} addTypename={false}>
        <OrganizationDetailContacts
          canEdit={true}
          organization={organization}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))
    expect(getByTestId(ASSIGN_BUTTON_TEST_ID)).toHaveTextContent('Assign')
    expect(container).toMatchSnapshot()
  })

  test('If input name is empty shouldn\'t add pill after clicking assign', async () => {
    const { container, getByTestId, queryByTestId } = render(
      <CustomMockedProvider mocks={[mockContacts]} addTypename={false}>
        <OrganizationDetailContacts
          canEdit={true}
          organization={organization}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))

    const input = getByTestId(NAME_INPUT_TEST_ID)

    fireEvent.change(input, { target: { value: '' } })
    await act(async () => fireEvent.click(getByTestId(ASSIGN_BUTTON_TEST_ID)))
    expect(queryByTestId(PILL_TEST_ID)).toBeNull()
    expect(container).toMatchSnapshot()
  })

  test('Should add pill after click assign button if input name has value', async () => {
    const { container, getByTestId } = render(
      <CustomMockedProvider mocks={[mockContacts]} addTypename={false}>
        <OrganizationDetailContacts
          canEdit={true}
          organization={organization}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))

    const inputName = getByTestId(NAME_INPUT_TEST_ID)
    const inputEmail = getByTestId(EMAIL_INPUT_TEST_ID)

    fireEvent.change(inputName, { target: { value: contacts.data.name } })
    fireEvent.change(inputEmail, { target: { value: contacts.data.email } })
    await act(async () => fireEvent.click(getByTestId(ASSIGN_BUTTON_TEST_ID)))
    await screen.findByTestId(PILL_TEST_ID)

    expect(getByTestId(PILL_TEST_ID)).toHaveTextContent(`Name: ${contacts.data.name}`)
    expect(container).toMatchSnapshot()
  })

  test('Should remove pill after click close icon on pill', async () => {
    const { container, getByTestId, queryByTestId } = render(
      <CustomMockedProvider mocks={[mockContacts]} addTypename={false}>
        <OrganizationDetailContacts
          canEdit={true}
          organization={organization}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))

    const inputName = getByTestId(NAME_INPUT_TEST_ID)
    const inputEmail = getByTestId(EMAIL_INPUT_TEST_ID)

    fireEvent.change(inputName, { target: { value: contacts.data.name } })
    fireEvent.change(inputEmail, { target: { value: contacts.data.email } })
    await act(async () => fireEvent.click(getByTestId(ASSIGN_BUTTON_TEST_ID)))
    await screen.findByTestId(PILL_TEST_ID)

    expect(getByTestId(PILL_TEST_ID)).toHaveTextContent(`Name: ${contacts.data.name}`)

    const removePillButton = getByTestId(PILL_REMOVE_BUTTON_TEST_ID)

    fireEvent.click(removePillButton)
    expect(queryByTestId(PILL_TEST_ID)).toBeNull()
    expect(container).toMatchSnapshot()
  })

  test('Should clear inputs after click assign button', async () => {
    const { container, getByTestId } = render(
      <CustomMockedProvider mocks={[mockContacts]} addTypename={false}>
        <OrganizationDetailContacts
          canEdit={true}
          organization={organization}
        />
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(EDIT_BUTTON_TEST_ID))

    const inputName = getByTestId(NAME_INPUT_TEST_ID)
    const inputEmail = getByTestId(EMAIL_INPUT_TEST_ID)
    const inputTitle = getByTestId(TITLE_INPUT_TEST_ID)

    fireEvent.change(inputName, { target: { value: contacts.data.name } })
    fireEvent.change(inputEmail, { target: { value: contacts.data.email } })
    fireEvent.change(inputTitle, { target: { value: contacts.data.title } })
    await act(async () => fireEvent.click(getByTestId(ASSIGN_BUTTON_TEST_ID)))

    expect(inputName).toHaveTextContent('')
    expect(inputEmail).toHaveTextContent('')
    expect(inputTitle).toHaveTextContent('')
    expect(container).toMatchSnapshot()
  })
})
