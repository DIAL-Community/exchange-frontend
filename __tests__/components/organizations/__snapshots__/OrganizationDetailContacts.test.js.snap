// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for the OrganizationDetailContacts component. If input name is empty shouldn't add pill after clicking assign 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    data-testid="organization-contacts"
  >
    <div
      class="mt-12"
      data-testid="editable-section"
    >
      <div
        class="flex justify-between items-center mb-3"
      >
        <div
          class="card-title text-dial-gray-dark"
        >
          Contacts
        </div>
        <div
          class="gap-3"
        />
      </div>
      <div
        class="bg-edit p-6"
      >
        <form>
          <p
            class="card-title text-dial-blue mb-3"
          >
            Assign
             
            Contacts
          </p>
          <div
            class="flex flex-col md:flex-row justify-between"
          >
            <label
              class="flex flex-col gap-y-2 mb-2 mx-4 w-full"
            >
              <p
                class="required-field"
              >
                Name:
              </p>
              <input
                class="validation-error w-full"
                data-testid="name-input"
                name="name"
                type="text"
                value=""
              />
              <p
                class="text-sm text-validation-error"
              >
                This field is required
              </p>
            </label>
            <label
              class="flex flex-col gap-y-2 mb-2 mx-4 w-full"
            >
              <p
                class="required-field"
              >
                Email:
              </p>
              <input
                class="validation-error w-full"
                data-testid="email-input"
                name="email"
                type="text"
                value=""
              />
              <p
                class="text-sm text-validation-error"
              >
                This field is required
              </p>
            </label>
            <label
              class="flex flex-col gap-y-2 mb-2 mx-4 w-full"
            >
              <p>
                Title:
              </p>
              <input
                class="w-full"
                data-testid="title-input"
                name="title"
                type="text"
                value=""
              />
            </label>
            <label
              class="flex md:block justify-center md:justify-end ml-4 mb-2 mt-8 text-xl"
            >
              <button
                class="submit-button"
                data-testid="assign-button"
                type="submit"
              >
                Assign
              </button>
            </label>
          </div>
          <div
            class="flex flex-wrap gap-3 mt-5 mx-4"
          />
        </form>
        <div
          class="flex justify-end mt-8 gap-3 text-xl"
        >
          <button
            class="submit-button"
            data-testid="submit-button"
            disabled=""
            type="submit"
          >
            Submit
          </button>
          <button
            class="cancel-button"
            data-testid="cancel-button"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit test for the OrganizationDetailContacts component. Should add pill after click assign button if input name has value 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    data-testid="organization-contacts"
  >
    <div
      class="mt-12"
      data-testid="editable-section"
    >
      <div
        class="flex justify-between items-center mb-3"
      >
        <div
          class="card-title text-dial-gray-dark"
        >
          Contacts
        </div>
        <div
          class="gap-3"
        />
      </div>
      <div
        class="bg-edit p-6"
      >
        <form>
          <p
            class="card-title text-dial-blue mb-3"
          >
            Assign
             
            Contacts
          </p>
          <div
            class="flex flex-col md:flex-row justify-between"
          >
            <label
              class="flex flex-col gap-y-2 mb-2 mx-4 w-full"
            >
              <p
                class="required-field"
              >
                Name:
              </p>
              <input
                class="w-full"
                data-testid="name-input"
                name="name"
                type="text"
                value=""
              />
            </label>
            <label
              class="flex flex-col gap-y-2 mb-2 mx-4 w-full"
            >
              <p
                class="required-field"
              >
                Email:
              </p>
              <input
                class="w-full"
                data-testid="email-input"
                name="email"
                type="text"
                value=""
              />
            </label>
            <label
              class="flex flex-col gap-y-2 mb-2 mx-4 w-full"
            >
              <p>
                Title:
              </p>
              <input
                class="w-full"
                data-testid="title-input"
                name="title"
                type="text"
                value=""
              />
            </label>
            <label
              class="flex md:block justify-center md:justify-end ml-4 mb-2 mt-8 text-xl"
            >
              <button
                class="submit-button"
                data-testid="assign-button"
                type="submit"
              >
                Assign
              </button>
            </label>
          </div>
          <div
            class="flex flex-wrap gap-3 mt-5 mx-4"
          >
            <div
              class="inline-flex items-center px-2 py-1 rounded-md bg-dial-blue-light text-dial-gray-dark"
              data-testid="pill"
            >
              Name: Test Test
              <svg
                class="ml-3 text-2xl cursor-pointer"
                data-testid="remove-button"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </div>
          </div>
        </form>
        <div
          class="flex justify-end mt-8 gap-3 text-xl"
        >
          <button
            class="submit-button"
            data-testid="submit-button"
            type="submit"
          >
            Submit
          </button>
          <button
            class="cancel-button"
            data-testid="cancel-button"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit test for the OrganizationDetailContacts component. Should assign button have text "Assign" 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    data-testid="organization-contacts"
  >
    <div
      class="mt-12"
      data-testid="editable-section"
    >
      <div
        class="flex justify-between items-center mb-3"
      >
        <div
          class="card-title text-dial-gray-dark"
        >
          Contacts
        </div>
        <div
          class="gap-3"
        />
      </div>
      <div
        class="bg-edit p-6"
      >
        <form>
          <p
            class="card-title text-dial-blue mb-3"
          >
            Assign
             
            Contacts
          </p>
          <div
            class="flex flex-col md:flex-row justify-between"
          >
            <label
              class="flex flex-col gap-y-2 mb-2 mx-4 w-full"
            >
              <p
                class="required-field"
              >
                Name:
              </p>
              <input
                class="w-full"
                data-testid="name-input"
                name="name"
                type="text"
                value=""
              />
            </label>
            <label
              class="flex flex-col gap-y-2 mb-2 mx-4 w-full"
            >
              <p
                class="required-field"
              >
                Email:
              </p>
              <input
                class="w-full"
                data-testid="email-input"
                name="email"
                type="text"
                value=""
              />
            </label>
            <label
              class="flex flex-col gap-y-2 mb-2 mx-4 w-full"
            >
              <p>
                Title:
              </p>
              <input
                class="w-full"
                data-testid="title-input"
                name="title"
                type="text"
                value=""
              />
            </label>
            <label
              class="flex md:block justify-center md:justify-end ml-4 mb-2 mt-8 text-xl"
            >
              <button
                class="submit-button"
                data-testid="assign-button"
                type="submit"
              >
                Assign
              </button>
            </label>
          </div>
          <div
            class="flex flex-wrap gap-3 mt-5 mx-4"
          />
        </form>
        <div
          class="flex justify-end mt-8 gap-3 text-xl"
        >
          <button
            class="submit-button"
            data-testid="submit-button"
            disabled=""
            type="submit"
          >
            Submit
          </button>
          <button
            class="cancel-button"
            data-testid="cancel-button"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit test for the OrganizationDetailContacts component. Should clear inputs after click assign button 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    data-testid="organization-contacts"
  >
    <div
      class="mt-12"
      data-testid="editable-section"
    >
      <div
        class="flex justify-between items-center mb-3"
      >
        <div
          class="card-title text-dial-gray-dark"
        >
          Contacts
        </div>
        <div
          class="gap-3"
        />
      </div>
      <div
        class="bg-edit p-6"
      >
        <form>
          <p
            class="card-title text-dial-blue mb-3"
          >
            Assign
             
            Contacts
          </p>
          <div
            class="flex flex-col md:flex-row justify-between"
          >
            <label
              class="flex flex-col gap-y-2 mb-2 mx-4 w-full"
            >
              <p
                class="required-field"
              >
                Name:
              </p>
              <input
                class="w-full"
                data-testid="name-input"
                name="name"
                type="text"
                value=""
              />
            </label>
            <label
              class="flex flex-col gap-y-2 mb-2 mx-4 w-full"
            >
              <p
                class="required-field"
              >
                Email:
              </p>
              <input
                class="w-full"
                data-testid="email-input"
                name="email"
                type="text"
                value=""
              />
            </label>
            <label
              class="flex flex-col gap-y-2 mb-2 mx-4 w-full"
            >
              <p>
                Title:
              </p>
              <input
                class="w-full"
                data-testid="title-input"
                name="title"
                type="text"
                value=""
              />
            </label>
            <label
              class="flex md:block justify-center md:justify-end ml-4 mb-2 mt-8 text-xl"
            >
              <button
                class="submit-button"
                data-testid="assign-button"
                type="submit"
              >
                Assign
              </button>
            </label>
          </div>
          <div
            class="flex flex-wrap gap-3 mt-5 mx-4"
          >
            <div
              class="inline-flex items-center px-2 py-1 rounded-md bg-dial-blue-light text-dial-gray-dark"
              data-testid="pill"
            >
              Name: Test Test
              <svg
                class="ml-3 text-2xl cursor-pointer"
                data-testid="remove-button"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </div>
          </div>
        </form>
        <div
          class="flex justify-end mt-8 gap-3 text-xl"
        >
          <button
            class="submit-button"
            data-testid="submit-button"
            type="submit"
          >
            Submit
          </button>
          <button
            class="cancel-button"
            data-testid="cancel-button"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit test for the OrganizationDetailContacts component. Should remove pill after click close icon on pill 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    data-testid="organization-contacts"
  >
    <div
      class="mt-12"
      data-testid="editable-section"
    >
      <div
        class="flex justify-between items-center mb-3"
      >
        <div
          class="card-title text-dial-gray-dark"
        >
          Contacts
        </div>
        <div
          class="gap-3"
        />
      </div>
      <div
        class="bg-edit p-6"
      >
        <form>
          <p
            class="card-title text-dial-blue mb-3"
          >
            Assign
             
            Contacts
          </p>
          <div
            class="flex flex-col md:flex-row justify-between"
          >
            <label
              class="flex flex-col gap-y-2 mb-2 mx-4 w-full"
            >
              <p
                class="required-field"
              >
                Name:
              </p>
              <input
                class="w-full"
                data-testid="name-input"
                name="name"
                type="text"
                value=""
              />
            </label>
            <label
              class="flex flex-col gap-y-2 mb-2 mx-4 w-full"
            >
              <p
                class="required-field"
              >
                Email:
              </p>
              <input
                class="w-full"
                data-testid="email-input"
                name="email"
                type="text"
                value=""
              />
            </label>
            <label
              class="flex flex-col gap-y-2 mb-2 mx-4 w-full"
            >
              <p>
                Title:
              </p>
              <input
                class="w-full"
                data-testid="title-input"
                name="title"
                type="text"
                value=""
              />
            </label>
            <label
              class="flex md:block justify-center md:justify-end ml-4 mb-2 mt-8 text-xl"
            >
              <button
                class="submit-button"
                data-testid="assign-button"
                type="submit"
              >
                Assign
              </button>
            </label>
          </div>
          <div
            class="flex flex-wrap gap-3 mt-5 mx-4"
          />
        </form>
        <div
          class="flex justify-end mt-8 gap-3 text-xl"
        >
          <button
            class="submit-button"
            data-testid="submit-button"
            type="submit"
          >
            Submit
          </button>
          <button
            class="cancel-button"
            data-testid="cancel-button"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
